triggers/service/authentication.md

```diff
@@ -0,0 +1,95 @@
+# Design for Trigger Service Authentication/Authorization
+
+## Goals
+
+- Be compatible with an OAuth2 authorization code grant
+  https://tools.ietf.org/html/rfc6749#section-4.1
+- Do not require OAuth2 or any other specific
+  authentication/authorization protocol from the IAM. In other words,
+  the communication with the IAM must be pluggable.
+- Do not rely on wildcard access for the trigger service, it should
+  only be able to start triggers on behalf of a party if a user that
+  controls that party has given consent.
```

Comment:
👏 curious about the mechanism for verifying the consent here: is it the grant itself, or some user metadata that was the result of a previous authentication step?

Reply:
That’s up to the middleware. One option would be that the middleware translates the required claims to OAuth2 claims and includes those when redirecting to `/authorize`.